jax/experimental/sparse_ops.py

```diff
@@ -403,6 +420,23 @@ def _coo_matvec_gpu_translation_rule(c, data, row, col, v, *, shape, transpose):
   xla.backend_specific_translations['gpu'][
       coo_matvec_p] = _coo_matvec_gpu_translation_rule
 
+def _coo_matvec_jvp_rule(primals_in, tangents_in, **params):
+  vals, rows, cols, vec = primals_in
+  sparse_mat_dot, _, _, vec_dot = tangents_in
+
+  primals_out = coo_matvec(vals, rows, cols, vec, **params)
+  is_zero = lambda x: type(x) is ad_util.Zero
+
+  if is_zero(sparse_mat_dot) and is_zero(vec_dot):
+    tangents_out = ad_util.Zero.from_value(primals_out)
+  elif not is_zero(sparse_mat_dot) and is_zero(vec_dot):
+    tangents_out = coo_matvec(sparse_mat_dot, rows, cols, vec, **params)
+  elif is_zero(sparse_mat_dot) and not is_zero(vec_dot):
+    tangents_out = coo_matvec(vals, rows, cols, vec_dot, **params)
+  else:
+    tangents_out = coo_matvec(sparse_mat_dot, rows, cols, vec, **params) + coo_matvec(vals, rows, cols, vec_dot, **params)
```

Comment:
I think we could simplify this with something like:
```python
_zero = lambda p, t: lax.zeros_like_array(p) if isinstance(t, ad.Zero) else t
data_dot = _zero(data, data_dot)
v_dot = _zero(v, v_dot)

tangents_out = (
  coo_matvec(data_dot, row, col, v, shape=shape, transpose=transpose) +
  coo_matvec(data, row, col, v_dot, shape=shape, transpose=transpose)
)
```

Reply:
I had something like this originally, but don't you run the risk of doing unnecessary ops if one of the tangents is zero? I changed it anyway - it's much prettier :-).